jscomp/others/js_null_undefined.ml

```diff
@@ -33,11 +33,11 @@ external empty : 'a t = "undefined" [@@bs.val]
 
 let bind x f =
   match to_opt x with
-  | None ->  undefined
-  | Some x -> return (f  x [@bs])
+  | None -> (Obj.magic (x: 'a t): 'b t)
```

Comment:
okay, this is more confusing, I understand why it works here, I will add type annotations to the parameter after merging : )